siteimprove/admin/js/siteimprove.js

```diff
@@ -170,7 +170,8 @@
       // 0 = overlay-v1.js
       // 1 = overlay-latest.js
       if (this.version == 1 && this.is_content_page) {
-        _si.push(['registerPrepublishCallback', getDomCallback, this.token]);
+        var cb = Boolean(+php_vars.has_api_key) ? getDomCallback : null
```

Comment:
For showing the prepublish button in the UI, we both check the "has api key" (now), but also check the `siteimprove_prepublish_allowed` and `siteimprove_prepublish_allowed`. I think it's an oversight that we don't check those here - there should be the same logic behind the button inside the side panel and the one in WP. Can you check if it's easy to add those two here as well? And maybe also trace back and understand what logic lies behind those two?

Reply:
New commit that fixes this.